### DISCUSSION TITLE
Round Table Discussion Setup

### DISCUSSION THREAD

**Discussion by jduke99** [langroid/langroid#551] [Q&A]
Hi there, love the library, thank you for your work.

I have been mostly doing projects with a primary agent who brings in various consultants and that all works well (adding subtasks to the main PM task).

But I am now trying to get a round table conversation with all of the parties talking whenever.  I do not want it to be sequential or constrained to each one only going after a particular other agent.  How would I set up the chat so that all agents may interject at any time?

Thanks!

**Comment by pchalasani** [comment]
Thanks @jduke99, glad you like the library. At the moment one way to do this is using the `run_task_batch_gen` in `batch.py`, e.g. something like this:

https://github.com/langroid/langroid/blob/main/examples/basic/multi-agent-round-table.py

More generally we are thinking about abstracting a "scheduler" that generalizes beyond the linear/sequential polling of agents.

(Just pushed a new version of this example)
